src/Inputs/Slider.tsx

```diff
@@ -1,79 +1,440 @@
+                    e.target === selectedBar.current
+                ) {
+                    setStartThumbLeft(calculatePosition(clickedOn));
+                }
+            }
+            document.addEventListener('mousemove', onMouseMove as any);
+            document.addEventListener('mouseup', onMouseUp);
+        }
+    };
+
+    return (
+        <LabelLayout {...props}>
+            <SliderBoard
+                ref={bar}
+                disabled={disabled}
+                onMouseDown={(event): void => handleMouseDown(event)}
+                theme={theme}
+            >
+                <SliderBoardSelected
+                    ref={selectedBar}
+                    left={startThumbLeft}
+                    right={finishThumbLeft - startThumbLeft}
+                    disabled={disabled}
+                    hasRail={hasRail}
+                    theme={theme}
+                />
+                {hasTwoInputs && (
+                    <SliderThumbStart
+                        ref={startThumb}
+                        left={startThumbLeft}
+                        disabled={disabled}
+                        hasTwoInputs={hasTwoInputs}
+                        onMouseDown={(event): void =>
+                            handleMouseDown(event as MouseEvent)
+                        }
+                    >
+                        {hasPopup && (
+                            <Popup
+                                top={popuptop}
+                                left={popupleft}
+                                width={popupwidth}
+                                height={popupheight}
+                            >
+                                {startThumbLeftInValue}
+                            </Popup>
+                        )}
+                    </SliderThumbStart>
+                )}
+
+                <SliderThumb
+                    ref={finishThumb}
+                    disabled={disabled}
+                    hasTwoInputs={hasTwoInputs}
+                    left={finishThumbLeft}
+                    onMouseDown={(event): any =>
+                        handleMouseDown(event as MouseEvent)
+                    }
+                >
+                    {hasPopup && (
+                        <Popup
+                            top={popuptop}
+                            left={popupleft}
+                            width={popupwidth}
+                            height={popupheight}
+                        >
+                            {finishThumbLeftInValue}
+                        </Popup>
+                    )}
+                </SliderThumb>
+            </SliderBoard>
+            {marks && (
+                <SliderBoardMarks
+                    ref={marksBar}
+                    theme={theme}
+                    disabled={disabled}
+                >
+                    {marks.map(
+                        ({ key, mark }): React.ReactElement => (
+                            <div
```

Comment:
Where is the key?

Reply:
Sorry I don't get this

Reply:
https://reactjs.org/docs/lists-and-keys.html